q2cli/builtin/tools.py

```diff
@@ -907,3 +908,249 @@ def cache_status(cache):
+        out_fp=out_fp,
+        validate_checksums=validate_checksums,
+        parse_metadata=parse_metadata,
+        recurse=recurse,
+        use_recorded_metadata=use_recorded_metadata,
+        suppress_header=suppress_header,
+        verbose=verbose,
+        dump_recorded_metadata=dump_recorded_metadata,
+        md_out_fp=metadata_out_fp
+    )
+    filename = os.path.realpath(out_fp)
+    click.echo(f'{usage_driver} replay script written to {filename}')
+
+
+@tools.command(name='replay-citations', cls=ToolCommand)
+@click.option('--in-fp', required=True,
+              help='filepath to a QIIME 2 Archive or directory of Archives')
+@click.option('--recurse/--no-recurse',
+              default=False,
+              show_default=True,
+              help=('if in-fp is a directory, will also search sub-directories'
+                    ' when finding .qza/.qzv files to parse'))
+@click.option('--deduplicate/--no-deduplicate',
+              default=True,
+              show_default=True,
+              help=('If deduplicate, duplicate citations will be removed '
+                    'heuristically, e.g. by comparing DOI fields. '
+                    'This greatly reduces manual curation of reference lists, '
+                    'but introduces a small risk of reference loss.'))
+@click.option('--suppress-header/--no-suppress-header',
+              default=False,
+              show_default=True,
+              help='do not write header/footer blocks in the output file')
+@click.option('--verbose/--no-verbose',
+              default=True,
+              show_default=True,
+              help='print status messages to stdout while processing')
+@click.option('--out-fp',
+              required=True,
+              help='the filepath where your bibtex file should be written.')
+def citations_replay(
+    in_fp: str,
+    out_fp: str,
+    recurse: bool = False,
+    deduplicate: bool = True,
+    suppress_header: bool = False,
+    verbose: bool = True
+):
+    """
+    Reports all citations from a QIIME 2 Artifact or directory of Artifacts,
+    with the goal of improving and simplifying attribution of/in published
+    work.
+
+    Not for use in reporting e.g. software versions used in an analysis, as
+    deduplication removes duplicate references with different plugin versions.
+    """
+    from qiime2.core.archive.provenance_lib.parse import ProvDAG
+    from qiime2.core.archive.provenance_lib.replay import replay_citations
+
+    dag = ProvDAG(in_fp, verbose=verbose, recurse=recurse)
+    replay_citations(
+        dag,
+        out_fp=out_fp,
+        deduplicate=deduplicate,
+        suppress_header=suppress_header
+    )
+    filename = os.path.realpath(out_fp)
+    click.echo(f'citations bibtex file written to {filename}')
+
+
+@tools.command(name='replay-supplement', cls=ToolCommand)
+@click.option('--in-fp', required=True,
+              help='filepath to a QIIME 2 Archive or directory of Archives')
```

Comment:
Same as other `--in-fp`, we could make this a [click.Path](https://github.com/qiime2/q2cli/blob/6c91089e0fad52b3558586f93cdb8f01b0379134/q2cli/builtin/dev.py#L78) exists=True, file_okay=True, dir_okay=True, readable=Tru